crates/wast/tests/parse-component.rs

```diff
@@ -0,0 +1,67 @@
+//! A test suite to parse everything in `parse-fail` and assert that it matches
+//! the `*.err` file it generates.
+//!
+//! Use `BLESS=1` in the environment to auto-update `*.err` files. Be sure to
+//! look at the diff!
+
+use rayon::prelude::*;
+use std::path::{Path, PathBuf};
+
+#[ignore] // TODO: Re-enable this once the rest of encoding is implemented.
+#[test]
+fn parse_component() {
+    let mut tests = Vec::new();
+    find_tests("tests/parse-component".as_ref(), &mut tests);
+    let filter = std::env::args().nth(1);
+
+    let tests = tests
+        .iter()
+        .filter(|test| {
+            if let Some(filter) = &filter {
+                if let Some(s) = test.file_name().and_then(|s| s.to_str()) {
+                    if !s.contains(filter) {
+                        return false;
+                    }
+                }
+            }
+            true
+        })
+        .collect::<Vec<_>>();
+
+    println!("running {} tests\n", tests.len());
+
+    let errors = tests
+        .par_iter()
+        .filter_map(|test| run_test(test).err())
+        .collect::<Vec<_>>();
+
+    if !errors.is_empty() {
+        for msg in errors.iter() {
+            eprintln!("{}", msg);
+        }
+
+        panic!("{} tests failed", errors.len())
+    }
+
+    println!("test result: ok. {} passed\n", tests.len());
+}
+
+fn run_test(test: &Path) -> anyhow::Result<()> {
+    let _ = wat::parse_file(test)?;
+    Ok(())
+}
+
+fn find_tests(path: &Path, tests: &mut Vec<PathBuf>) {
+    for f in path.read_dir().unwrap() {
+        let f = f.unwrap();
+        if f.file_type().unwrap().is_dir() {
+            find_tests(&f.path(), tests);
+            continue;
+        }
+        match f.path().extension().and_then(|s| s.to_str()) {
+            Some("wat") => {}
+            _ => continue,
+        }
+        tests.push(f.path());
+    }
+}
```

Comment:
Could all of these tests be added to `tests/roundtrip.rs` and/or something like `tests/local/component-model`? They ideally would all exercise wasmparser's validation and wasmprinter's printing for round-trip-ness as well.

Reply:
Yes, that will be good to do once everything roundtrips, but wasmparser doesn't appear to support everything that this new wast encoder is encoding yet. I'll move things to roundtrip tests once we work these out.